### PR TITLE
Implement DATE_TRUNC by day

### PR DESCRIPTION
**Pull request by marcellustavares**
hi @goccy, I've implemented the DATE_TRUNC by day case.

Thanks

### CODE REVIEW

File: internal/function_date_test.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package internal
+
+import (
+	"testing"
+	"time"
+)
+
+func TestDateTrunc(t *testing.T) {
```
Comment (goccy) [inline internal/function_date_test.go:8]: Please add a query based test case to query_test.go to test the `DATE_TRUNC/DAY` function.
If there is the test case, we don't need this one.

Comment (marcellustavares) [reply internal/function_date_test.go:8]: 👍

Comment (marcellustavares) [reply internal/function_date_test.go]: done @goccy, thanks!

**Comment by goccy** [comment]
Thank you for the great PR !! LGTM

**Pull request merged**